### PR TITLE
Drop support for EOL Python 2.6, 3.2 and 3.3

### LINKED ISSUE #295: Drop support for Python 2.6 and 3.2 and 3.3
Those versions are really old. @hgrecco do you have an opinion

### PR DESCRIPTION
**Pull request by hugovk**
Fixes #295.

### CODE REVIEW

**Comment by hugovk** [comment]
```
branches:
  only:
```
This section in .travis.yml is an impediment to contributions because it means we can't test our feature branches before creating PRs, which might cause undue noise if a branch isn't yet ready for a PR.

**Comment by MatthieuDartiailh** [comment]
ping @hgrecco 

In the absence of answer I will merge at the end of the month. @hugovk the restriction on the branches is their because otherwise teh temporary branches used by bors causes issue. I honestly do not mind an early PR as it is nice to know people are working on something. You can put [WIP] to let maintainers know that it is not ready and that's fine.

**Comment by MatthieuDartiailh** [comment]
As nobody pronounced against it and since we are not enough maintainers to assume the burden of those old versions, I will merge. Thanks @hugovk 

bors r+

File: setup.py [hunk 1]
```diff
@@ -32,9 +32,6 @@ def read(filename):
 if sys.version_info < (3, 4):
```
Comment (PeterPablo) [inline setup.py]: shouldn't this also be removed, since support now starts at >= 3.4?

Comment (MatthieuDartiailh) [reply setup.py]: Indeed thanks for pointing it out.

Comment (hugovk) [reply setup.py]: What about Python 2.7?

Comment (MatthieuDartiailh) [reply setup.py]: This one has not reached EOL and is still widely used. So I don't believe, we (pyvisa maintainers) will drop it before 2020-2021 or even a bit later. That said I won't fight to keep it longer than necessary as it increases the maintenance burden.

Comment (hugovk) [reply setup.py]: Yes, still big numbers for 2.7 at https://github.com/pyvisa/pyvisa/issues/295#issuecomment-349346466. 

Python 2.7 doesn't have the `enum` module, but this `enum34` package backports it and is needed for 2.7.

It's probably not worth dropping 2.7 just to remove this dependency.

Comment (MatthieuDartiailh) [reply setup.py]: I should have been more careful in checking that. I reverted my commit. Sorry @hugovk.

Comment (MatthieuDartiailh) [reply setup.py]: I may edit the check later to avoid mentioning Python 3.4 to avoid future confusions.

Comment (hugovk) [reply setup.py]: Good idea, a simple Py2 vs. Py3 check would be better!

**Pull request merged**